styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -309,6 +318,53 @@ private void processInstance(StyxConfig config, Map<String, Resource> resources,
     sendDequeue(instance, runState, instanceResourceRefs);
   }
 
+  /**
+   * Shuffle instances. For the same workflow ID, older instances will appear early in the list.
+   */
+  @VisibleForTesting
+  List<WorkflowInstance> shuffleInstances(Set<WorkflowInstance> activeInstances) {
+    var groups = activeInstances
+        .stream()
+        .collect(groupingBy(WorkflowInstance::workflowId, LinkedHashMap::new,
+            toCollection(() -> new TreeSet<>(comparing(WorkflowInstance::parameter)))));
+
+    var shuffledGroups = new ArrayList<SortedSet<WorkflowInstance>>(groups.values());
+    shuffler.shuffle(shuffledGroups);
```

Comment:
this shuffle seems unnecessary now with the shuffle in the shuffleInstances method

Reply:
Maybe or maybe not. I guess it doesn't hurt with more randomness.

Reply:
I am with babis here, the shuffle at merge is directing which workflow goes to what index, a previous shuffle won't improve the randomness here, so it is unnecessary

Reply:
👍